.github/workflows/lint-and-test.yaml

```diff
@@ -2,50 +2,138 @@ name: Lint and test PR
 
 on: pull_request
 
+# Prevent to consume free CI minutes by not running two workflows in the CI at the same time.
+concurrency:
+  group: pr-testing
+  cancel-in-progress: true
+
 jobs:
-  need-test:
+  docker-build:
+    name: Build PowerDNS image
     runs-on: ubuntu-latest
-    outputs:
-      charts: ${{ steps.list-changed.outputs.changed }}
     steps:
       - uses: actions/checkout@v3
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Build
+        uses: docker/build-push-action@v4
         with:
-          fetch-depth: 0
-      - uses: helm/chart-testing-action@v2.3.1
-      - id: list-changed
-        run: |
-          changed=$(ct --config .github/ct.yaml list-changed)
-          if [[ -n "$changed" ]]; then
-            echo "changed=true" >> $GITHUB_OUTPUT
-          fi
+          tags: ${{ github.repository }}-ci
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
 
-  lint-chart:
+  lint-charts:
+    name: Lint charts
     runs-on: ubuntu-latest
-    needs: need-test
-    if: ${{ needs.need-test.outputs.charts == 'true' }}
+    # Wait for docker-build job to complete to reuse build cache.
+    needs: [ docker-build ]
     steps:
-      - uses: actions/checkout@v3
+      - name: Checkout code
+        uses: actions/checkout@v3
         with:
           fetch-depth: 0
-      - uses: helm/chart-testing-action@v2.3.1
-      - run: helm plugin install https://github.com/helm-unittest/helm-unittest
-      - uses: manusa/actions-setup-minikube@v2.7.2
+      - name: Install helm/chart-testing
+        uses: helm/chart-testing-action@v2.3.1
+
+      # Lint charts and run unit tests
+      - name: Lint charts
+        run: ct --config .github/ct.yaml lint --all
+
+      - name: Install helm unit test siute
+        run: helm plugin install https://github.com/helm-unittest/helm-unittest
+      - name: Run unit tests
+        run: |
+          for chart in $(ct list-changed --config .github/ct-lint.yaml); do
+            if [ -d "$chart/tests/" ]; then
+              helm unittest $chart -3
+            fi
+          done
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Build image
+        uses: docker/build-push-action@v4
+        with:
+          tags: ${{ github.repository }}-ci
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+
+      # Install Minikube to test install paths
+      - name: Install Minikube
+        uses: manusa/actions-setup-minikube@v2.7.2
         with:
           minikube version: v1.29.0
-          kubernetes version: v1.25.0
+          kubernetes version: v1.26.0
           github token: ${{ secrets.GITHUB_TOKEN }}
+          driver: docker
+          start args: "--container-runtime=containerd"
```

Comment:
If we need to set up minikube for both linting and E2E, I'm starting to think it may make sense to squash these two jobs into one. Thoughts @kang-makes ?